expressions/object-literals.md

```diff
@@ -104,21 +128,22 @@ actor Main
     let l = List[String]
     l.push("hello").push("world")
     var count = U32(0)
-    for_each(l, lambda ref(s:String)(env,count) =>
-                    env.out.print(s)
-                    count = count + 1
-                end)
+    for_each(l, {ref(s:String)(env,count) =>
+      env.out.print(s)
+      count = count + 1
+    })
     // Displays '0' as the count
     env.out.print("Count: " + count.string())
 
-  fun for_each(l: List[String], f: {ref(String)}) =>
+  fun for_each(l: List[String], f: {ref(String)} ref) =>
     try
       f(l.shift())
       for_each(l, f)
     end
 ```
 
-This example declares the type of the apply function that is generated by the lambda expression as being `ref`. The type declaration for it in the `for_each` method also declares it as `ref`. The lambda function captures some variables so the object that is generated is `ref` and the default for the type declaration is `ref` so everything type checks.
+This example declares the type of the apply function that is generated by the 
+lambda expression as being `ref`. The lambda type declaration for the `f` parameter in the `for_each` method also declares it as `ref`. The reference capability of the lambda type must also be `ref` so that the method can be called. The lambda object does not need to declare an explicit reference capability because `val` is the default for a lambda that has captures.
```

Comment:
Shouldn't the last sentence say that `ref` is the default for a lambda that has captures?